app/vlinsert/main.go

```diff
@@ -29,15 +36,27 @@ func Stop() {
 func RequestHandler(w http.ResponseWriter, r *http.Request) bool {
 	path := r.URL.Path
 
-	if path == "/internal/insert" {
+	switch {
+	case strings.HasPrefix(path, "/internal/insert/") || path == "/internal/insert":
+		if *disableInternal {
+			httpserver.Errorf(w, r, "requests to /internal/insert/* are disabled with -internalinsert.disable command-line flag")
+			return true
+		}
+		fallthrough
+	case strings.HasPrefix(path, "/insert/"):
+		if *disableInsert {
+			httpserver.Errorf(w, r, "requests to /insert/* are disabled with -insert.disable command-line flag")
+			return true
+		}
+	default:
```

Comment:
It is better from readability and maintainability PoV wrapping all the code below for handling the `/insert/*` paths into a separate function, and call this function directly from here.